ansible/roles/ci/doc/how_does_it_work.md

```diff
@@ -0,0 +1,12 @@
+# How does it work
+
+The build tools was created to run on the [Docker](https://www.docker.com/) based CI servers.
+The Docker image for all build is one hosted on [Docker Hub](https://hub.docker.com/r/shadowrobot/ubuntu-ros-indigo-build-tools)
+
+The build script is loading Ubuntu 14.04 Docker image with pre-installed ROS Indigo.
+This image is dependent on Ubuntu and ROS repositories in Docker Hub which means that image is rebuild as soon as any of parent images changes.
+
+The build tools also put special marker in the image which indicates that ROS was installed using their script and workspace was created as well.
+Supported CI servers download latest version of the Docker image and run Ansible build tools inside.
+Number of modules can be used based on the preferences of the developers.
```

Comment:
The build tools also put a special marker in the image, which indicates that ROS was installed using their script and the workspace was created as well.
The supported CI servers download the latest version of the Docker image and run Ansible build tools inside.
The number of modules can be changed(?) based on the preferences of the developers.